app/models/katello/concerns/host_managed_extensions.rb

```diff
@@ -63,10 +63,42 @@ def content_and_puppet_match?
       end
 
       def import_package_profile(simple_packages)
-        self.installed_packages.where("nvra not in (?)", simple_packages.map(&:nvra)).destroy_all
-        existing_nvras = self.installed_packages.pluck(:nvra)
+        found = []
+
         simple_packages.each do |simple_package|
-          self.installed_packages.create!(:name => simple_package.name, :nvra => simple_package.nvra) unless existing_nvras.include?(simple_package.nvra)
+          ::Katello::Util::Support.active_record_retry do
+            pkg = InstalledPackage.where(:nvra => simple_package.nvra, :name => simple_package.name).limit(1)[0]
```

Comment:
you may want to add a comment just so it doesn't get changed to `.first` as part of a later refactor. It would be tricky to figure out what happened if someone did that and things started getting slow.